sql/pre_install/tables.sql

```diff
@@ -404,6 +404,18 @@ CREATE TABLE _timescaledb_catalog.continuous_aggs_invalidation_threshold (
 
 SELECT pg_catalog.pg_extension_config_dump('_timescaledb_catalog.continuous_aggs_invalidation_threshold', '');
 
+CREATE TABLE _timescaledb_catalog.continuous_aggs_watermark (
```

Comment:
Idea (feel free to ignore): Can't we simply create a bigserial for a cagg to track its watermark? We would have one bigserial for each cagg in the internal schema (next to the materialized hypertable itself).

That would avoid all the extra plumbing with catalog tables, indexes, etc., and we would benefit from internal (rel)caches. Plus we could just use `setval()` and `pg_sequence_last_value()` to update and read the watermark.

Reply:
The problem of sequences is they're independent of transaction. Imagine that for some reason the materialization transaction of refresh cagg fail, in this case the watermark will move forward but the data will not be materialized... not sure if a sequence is a good idea here!